src/test/scala/chiselTests/PrintableSpec.scala

```diff
@@ -261,4 +261,208 @@ class PrintableSpec extends AnyFlatSpec with Matchers {
+      case e                                                                                       => fail()
+    }
+  }
+  it should "handle printing ports of submodules with cf format specifier" in {
+    class MySubModule extends Module {
+      val io = IO(new Bundle {
+        val fizz = UInt(32.W)
+      })
+    }
+    class MyModule extends BasicTester {
+      val myInst = Module(new MySubModule)
+      printf(cf"${myInst.io.fizz}")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("%d", Seq("myInst.io.fizz"))) =>
+      case e                                        => fail()
+    }
+  }
+  it should "print UInts and SInts as Decimal by default with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myUInt = WireDefault(0.U)
+      val mySInt = WireDefault(-1.S)
+      printf(cf"$myUInt & $mySInt")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("%d & %d", Seq("myUInt", "mySInt"))) =>
+      case e                                               => fail()
+    }
+  }
+  it should "print Vecs like Scala Seqs by default with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myVec = Wire(Vec(4, UInt(32.W)))
+      myVec.foreach(_ := 0.U)
+      printf(cf"$myVec")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("Vec(%d, %d, %d, %d)", Seq("myVec[0]", "myVec[1]", "myVec[2]", "myVec[3]"))) =>
+      case e                                                                                       => fail()
+    }
+  }
+  it should "print Bundles like Scala Maps by default with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myBun = Wire(new Bundle {
+        val foo = UInt(32.W)
+        val bar = UInt(32.W)
+      })
+      myBun.foo := 0.U
+      myBun.bar := 0.U
+      printf(cf"$myBun")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("AnonymousBundle(foo -> %d, bar -> %d)", Seq("myBun.foo", "myBun.bar"))) =>
+      case e                                                                                   => fail()
+    }
+  }
+
+  it should "print regular scala variables with cf format specifier" in {
+
+    class MyModule extends BasicTester {
+      val f1 = 20.45156
+      val i1 = 10
+      val s1: Short = 15
+      val l1: Long = 253
+      printf(cf"F1 = $f1 D1 = $i1%2.4f F1 formatted = $f1%2.2f s1 = $s1%2.2f l1 = $l1%3.6f")
+
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("F1 = 20.45156 D1 = 10.0000 F1 formatted = 20.45 s1 = 15.00 l1 = 253.000000", Seq())) =>
+      case e                                                                                                => { println("e = ", e); fail() }
+    }
+  }
```

Comment:
I'm a little confused why you're only checkin %f, what about other format specifiers like `%d` and `%x`?

Reply:
Part of the grander test plan :) - Started with known challenging cases which I had to deal with during development.

Reply:
New test cases being developed - please do not review the spec file just as yet.